yt_dlp/extractor/jiosaavn.py

```diff
@@ -1,93 +1,120 @@
+import functools
+
 from .common import InfoExtractor
 from ..utils import (
+    format_field,
     int_or_none,
     js_to_json,
-    orderedSet,
+    make_archive_id,
+    url_basename,
     url_or_none,
     urlencode_postdata,
-    urljoin,
 )
 from ..utils.traversal import traverse_obj
 
 
 class JioSaavnBaseIE(InfoExtractor):
-    def _extract_initial_data(self, url, audio_id):
-        webpage = self._download_webpage(url, audio_id)
+    _VALID_BITRATES = ('16', '32', '64', '128', '320')
+
+    @functools.cached_property
+    def requested_bitrates(self):
+        requested_bitrates = self._configuration_arg('bitrate', ['128', '320'], ie_key='JioSaavn')
+        if invalid_bitrates := [br for br in requested_bitrates if br not in self._VALID_BITRATES]:
+            raise ValueError(
+                f'Invalid bitrate(s): {", ".join(invalid_bitrates)}. '
+                + f'Valid bitrates are: {", ".join(self._VALID_BITRATES)}')
```

Comment:
Why `ValueError`? Won't it behave like `expected=False`?
```suggestion
            raise ExtractorError(
                f'Invalid bitrate(s): {", ".join(invalid_bitrates)}. '
                f'Valid bitrates are: {", ".join(self._VALID_BITRATES)}', expected=True)
```

Reply:
that's what old code did, but agreed this is better

Reply:
> Won't it behave like `expected=False`?

actually, no it doesn't

`ValueError`:
```
[jiosaavn:song] OQsEfQFVUXk: Downloading webpage
ERROR: Invalid bitrate(s): 999. Valid bitrates are: 16, 32, 64, 128, 320
```

vs

`ExtractorError(expected=True)`
```
[jiosaavn:song] OQsEfQFVUXk: Downloading webpage
ERROR: [jiosaavn:song] OQsEfQFVUXk: Invalid bitrate(s): 123. Valid bitrates are: 16, 32, 64, 128, 320
```

I'd even argue the former is more appropriate and more readable for the user